codice/ddf
# DDF-2050 Added product data to ingest/dump commands so the entire catalog can be imported/exported.

#### What does this PR do?
1) Adds product data to ingest/dump commands so the entire catalog can be imported/exported.
2) Adds an "--include-content" option for content data in import/dump
3) Adds Security Logging to ingest/dump commands.
4) Adds some clean up to the catalog-core-commands package.
5) Adds an interface for InputStream -> List<Metacards>
6) Adds a ZipCompression, ZipDecompression and ZipValidator class for Zip file support
7) catalog:ingest now considers all metacards (history, etc).
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@jrnorth
@glenhein 
@dcruver 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@jlcsmith
@kcwire  
#### How should this be tested?
Build and run ddf.
Run catalog:dump with the "--include-content" option and observe local product data is dumped into a zip file.
Sign the zip with jarsigner and the localhost alias. 
Run catalog:ingest with the "--include-content" option and observe local product data is ingested into the catalog.
Attempt to download a resource to verify the correct linkage between content and metacard.
#### Any background context you want to provide?
#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-2050
#### Screenshots (if appropriate)
#### Checklist:
